docs: Revise README.md for Kaito inference (#581)
Azure/kaito: docs/inference/README.md

inference:
  preset:
    name: "falcon-7b"
```

> If a user runs Kaito in an on-premise Kubernetes cluster where GPU SKUs are unavailable, the GPU nodes can be prepared in advance. The user should ensure the corresponding vendor specific GPU plugin is installed successfully in every prepared node, i.e., the node status should report a non-zero allocatable GPU resource count. For example,

Reviewer: ```suggestion
If a user runs Kaito in an on-premise Kubernetes cluster where GPU SKUs are unavailable, the GPU nodes can be pre-configured. The user should ensure that the corresponding vendor-specific GPU plugin is successfully installed on each pre-configured node. The node status should report a non-zero GPU resource in the allocatable field. For example:
```

> If a user runs Kaito in an on-premise Kubernetes cluster where GPU SKUs are unavailable, the GPU nodes can be prepared in advance. The user should ensure the corresponding vendor specific GPU plugin is installed successfully in every prepared node, i.e., the node status should report a non-zero allocatable GPU resource count. For example,

Reviewer: ```suggestion
If a user runs Kaito in an on-premise Kubernetes cluster where GPU SKUs are unavailable, the GPU nodes can be pre-configured. The user should ensure that the corresponding vendor-specific GPU plugin is installed successfully in every prepared node, i.e. the node status should report a non-zero GPU resource in the allocatable field. For example:
```

        name: "falcon-7b-adapter"
        image:  "<YOUR_IMAGE>"
      strength: "0.2"
```

> Currently, only images are supported as adapter sources. The `strength` field specifies the multiplier for applying the adapter weights to the raw model weights.
> 
> Note: if you build a container image for an existing adapter, please copy all adapter files to the **/data** directory inside the container.
> 
> The detailed `InferenceSpec` API definitions can be found [here](https://github.com/Azure/kaito/blob/2ccc93daf9d5385649f3f219ff131ee7c9c47f3e/api/v1alpha1/workspace_types.go#L75).

Reviewer: ```suggestion
Currently, only images are supported as adapter sources. The `strength` field specifies the multiplier applied to the adapter weights relative to the raw model weights.

**Note:** When building a container image for an existing adapter, ensure all adapter files are copied to the **/data** directory inside the container.

For detailed `InferenceSpec` API definitions, refer to the [documentation](https://github.com/Azure/kaito/blob/2ccc93daf9d5385649f3f219ff131ee7c9c47f3e/api/v1alpha1/workspace_types.go#L75).
```

> The detailed `InferenceSpec` API definitions can be found [here](https://github.com/Azure/kaito/blob/2ccc93daf9d5385649f3f219ff131ee7c9c47f3e/api/v1alpha1/workspace_types.go#L75).

Reviewer: Use main branch for github link?

> # Inference workload
> 
> Depending on whether the specified model supports distributed inference or not, the Kaito controller will choose to use either Kubernetes **apps.deployment** workload (by default) or Kubernetes **apps.statefulset** workload (if the model supports distributed inference) to manage the inference service, which is exposed using a Cluster-IP type of Kubernetes `service`.
> 
> When adapters are specified in the `inference` spec, the Kaito controller will add initcontainers for every adapter in addition to the main container. The pod structure is illustrated in Figure 1.
> 
> <div align="left">
>   <img src="../img/kaito-inference-adapter.png" width=40% title="Kaito inference adapter" alt="Kaito inference adapter">
> </div>
> Figure 1. Kaito inference pod (with adapters) structure.
> 
> If an image is specified in the adapter source, the corresponding initcontainer uses the specified image as the container image. These initcontainers ensures all adapter data are available locally before the inference service starts. The main container uses one of the supported model images. The image entry launches the [inference\_api.py](../../presets/inference/text-generation/inference_api.py) script.
> 
> All containers use shared local volumes (by mounting the same EmptyDir volumes), hence file copies between containers are avoided.

Reviewer: ```suggestion
# Inference workload

The Kaito controller selects the appropriate Kubernetes workload type based on the model’s support for distributed inference:
- apps.deployment (default) for models not supporting distributed inference.
- apps.statefulset for models that support distributed inference.

The inference service is exposed using a Cluster-IP type Kubernetes `service`.

When adapters are specified in the `inference` spec, the Kaito controller adds an initcontainer for each adapter in addition to the main container. The pod structure is shown in Figure 1.

<div align="left">
  <img src="../img/kaito-inference-adapter.png" width=40% title="Kaito inference adapter" alt="Kaito inference adapter">
</div>

If an image is specified as the adapter source, the corresponding initcontainer uses that image as its container image. These initcontainers ensure all adapter data is available locally before the inference service starts. The main container uses a supported model image, launching the [inference_api.py](https://github.com/Azure/kaito/presets/inference/text-generation/inference_api.py) script.

All containers share local volumes by mounting the same `EmptyDir` volumes, avoiding file copies between containers.
```

> ## Workload update
> 
> Users can update the `workspace` customer resource and change the `adapters` field in the inference spec. The Kaito controller will apply a workload deployment update to make adapter changes take effect. The inference service pod will be recreated, causing a short period of service downtime. Once the new adapters are merged with the raw model weights and loaded into the GPU memory, the service will resume.

Reviewer: ```suggestion
## Workload update

To update the `adapters` field in the `inference` spec, users can modify the `workspace` custom resource. The Kaito controller will apply the changes, triggering a workload deployment update. This will recreate the inference service pod, resulting in a brief service downtime. Once the new adapters are merged with the raw model weights and loaded into GPU memory, the service will resume.
```
